libraries/engage/core/hocs/withCurrentProduct.jsx

```diff
@@ -21,25 +22,26 @@ const getInjectedProps = (context, prop) => {
  * Injects the current Product Context information into the desired component.
  * @param {Function} WrappedComponent The react component to wrap.
  * @param {Object} [options] A optional prop name to inject the current product properties.
- * @param {string} options.prop An optional prop name to inject the current product properties.
+ * @property {string} [options.prop] An optional prop name to inject the current product properties.
  * @returns {JSX}
  */
-export function withCurrentProduct(WrappedComponent, options = {}) {
+export function withCurrentProduct<P = {}>(
```

Comment:
This HOC is unexpected in core domain. And is strange. Is it even used somewhere except locations domain ?

Reply:
True, but it already exists since quite a while now and could be used in extension :/